stdlib/public/Darwin/Accelerate/Quadrature.swift

```diff
@@ -0,0 +1,287 @@
+        integrateOptions.rel_tolerance = relativeTolerance
+        
+        switch integrator {
+        case .qng:
+            integrateOptions.integrator = QUADRATURE_INTEGRATE_QNG
+        case .qag(let pointsPerInterval, let maxIntervals):
+            integrateOptions.integrator = QUADRATURE_INTEGRATE_QAG
+            integrateOptions.qag_points_per_interval = pointsPerInterval.points
+            integrateOptions.max_intervals = maxIntervals
+        case .qags(let maxIntervals):
+            integrateOptions.integrator = QUADRATURE_INTEGRATE_QAGS
+            integrateOptions.max_intervals = maxIntervals
+        }
+    }
+    
+    /// The quadrature instance's requested absolute tolerance on the result.
+    public var absoluteTolerance: Double {
+        set {
+            integrateOptions.abs_tolerance = newValue
+        }
+        get {
+            return integrateOptions.abs_tolerance
+        }
+    }
+    
+    /// The quadrature instance's requested relative tolerance on the result.
+    public var relativeTolerance: Double {
+        set {
+            integrateOptions.rel_tolerance = newValue
+        }
+        get {
+            return integrateOptions.rel_tolerance
+        }
+    }
+    
+    /// Performs the integration over the supplied function.
+    ///
+    /// - Parameter interval: The lower and upper bounds of the integration interval.
+    /// - Parameter integrand: The function to integrate. The input value is `x` that's within the interval over which the integrand is being integrated, and the output value is the corresponding value `y = integrand(x)` at those points.
+    public func integrate<U, V>(over interval: ClosedRange<Double>,
+                                integrand: (_ input: U, _ result: V) -> ()) ->
+        Result<(integralResult: Double, estimatedAbsoluteError: Double), Error>
+        where
+        U: _ContiguousCollection,
+        V: _MutableContiguousCollection,
+        U.Element == Double,
+        V.Element == Double {
+            
+            var status = QUADRATURE_SUCCESS
+            var estimatedAbsoluteError: Double = 0
+            var result: Double = 0
+            
+            var callback: quadrature_integrate_function!
+            
+            withoutActuallyEscaping(integrand) {escapableIntegrand in
+                withUnsafePointer(to: escapableIntegrand) {
+                    let integrandPointer = UnsafeMutableRawPointer(mutating: $0)
+                    
+                    callback = quadrature_integrate_function(
+                        fun: { (arg: UnsafeMutableRawPointer?,
+                            n: Int,
+                            x: UnsafePointer<Double>,
+                            y: UnsafeMutablePointer<Double>
+                            ) in
+                            
+                            guard let integrand = arg?.load(as: ((UnsafeBufferPointer<Double>, UnsafeMutableBufferPointer<Double>) ->()).self) else {
```

Comment:
Yikes! This is treating a closure of type `(U, V)->()` as a `(UBP<Double>, UMBP<Double>)->()`. Since the caller decides what they want U/V to be, that will be undefined behaviour if they choose some other type, like `Array<Double>`. I don't think this can be made generic.

Since this is overload is for batched processing, the element-wise version could be refactored to use it:
```swift
// <better name goes here>
func integrateVec(over interval: ClosedRange<Double>, integrand: (UBP<Double>, UMBP<Double>)->Void) -> Result<...> {
  // ...
}

func integrate(over interval: ClosedRange<Double>, integrand: (Double, inout Double)->Void) -> Result<...> {
  return integrateVec(over: interval) { src, dst in
    for i in 0..<src.count { integrand(src[i], dst[i]) } 
  }
}
```

Also, since the element-wise version is a convenience now, I personally think it would be _most_ convenient if the integrand had the signature `(Double)->Double`. There's no reason to read from the result buffer, and that's the kind of signature most mathematical functions you're likely to see in Swift will have.

Reply:
Indeed, yes, trying to generic-ize that function wasn't the best thing to do. I've changed the integrand signature to `integrand: (UBP<Double>, UMBP<Double>)->Void`. (pending push)

Reply:
As for the name: sure. I figured it might be ambiguous which one you meant in certain contexts, but it seems I misread the integrand signatures. They have different enough shapes that I think it shouldn't cause any difficulties in practice.

Reply:
Thanks. My change to `integrand: (UBP<Double>, UMBP<Double>)->Void` is available.